core/src/main/scala/sttp/tapir/model/ServerRequest.scala

```diff
@@ -16,6 +17,7 @@ trait ServerRequest extends RequestMetadata {
   def queryParameters: QueryParams
 
   lazy val acceptsContentTypes: Either[String, Seq[ContentTypeRange]] = Accepts.parse(headers)
+  val requestStart: Deadline = Deadline.now
```

Comment:
maybe instead of adding this here, we could use a `RequestInterceptor`, capture the starting timestamp there, and create an `EndpointInterceptor` with it?

Reply:
That should work, if this `RequestInterceptor` would be head of interceptors list, then `EndpointInteceptor` created by it would be called last which is what we need for metrics. 
But `requestStart` would have to be added as next argument to `Metrics` callback.